File: app/src/main/java/kaufland/com/demo/example/EasyExample.java
package kaufland.com.demo.example;

import android.app.Fragment;
import android.os.Bundle;
import androidx.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import kaufland.com.demo.R;
import kaufland.com.swipelibrary.DragView;
import kaufland.com.swipelibrary.SwipeLayout;
import kaufland.com.swipelibrary.SwipeState;

/**
 * Created by sbra0902 on 30.03.17.
 */

public class EasyExample extends Fragment {


    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, Bundle savedInstanceState) {
        return inflater.inflate(R.layout.easy_example_fragment, container, false);
    }

    @Override
    public void onViewCreated(final View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        setupExample2(view);

        final TextView mQuantityTextView = (TextView) view.findViewById(R.id.txt_quantity);

        view.findViewById(R.id.btn_add).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mQuantityTextView.setText(String.valueOf(Integer.valueOf(mQuantityTextView.getText().toString()) +1));
            }
        });

        view.findViewById(R.id.btn_minus).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mQuantityTextView.setText(String.valueOf(Integer.valueOf(mQuantityTextView.getText().toString()) -1));
            }
        });


    }

    private void setupExample2(final View view) {
        final SwipeLayout mViewById = (SwipeLayout) view.findViewById(R.id.swipe_example_2);
        mViewById.setSwipeListener(new SwipeLayout.SwipeListener() {
            @Override
            public void onSwipeOpened(SwipeState.DragViewState openedDragView, boolean isFullSwipe) {
                Toast.makeText(getActivity(), "onSwipeOpened " + openedDragView.name() + String.valueOf(isFullSwipe), Toast.LENGTH_SHORT).show();
            }

            @Override
            public void onSwipeClosed(SwipeState.DragViewState dragViewState) {
                Toast.makeText(getActivity(), "onSwipeOpened " + dragViewState.name(), Toast.LENGTH_SHORT).show();
            }

            @Override
            public void onBounce(SwipeState.DragViewState dragViewState) {
                Toast.makeText(getActivity(), "onBounce " + dragViewState.name(), Toast.LENGTH_SHORT).show();
            }
        });
        view.findViewById(R.id.btn_close).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mViewById.closeSwipe();
            }
        });

        view.findViewById(R.id.surface_view_example2).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Toast.makeText(getActivity(), "Clicked", Toast.LENGTH_SHORT).show();
            }
        });
        view.findViewById(R.id.btn_open_left).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mViewById.openSwipe(SwipeLayout.Companion.getLEFT_DRAG_VIEW());
            }
        });
        view.findViewById(R.id.btn_open_right).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mViewById.openSwipe(SwipeLayout.Companion.getRIGHT_DRAG_VIEW());
            }
        });
        view.findViewById(R.id.btn_disable_left_drag).setOnClickListener(new View.OnClickListener() {
            private boolean state = true;
            public void onClick(View v) {
                if ( state ) {
                    state = false;
                    ((Button)v).setText("enable left drag");

                } else {
                    state = true;
                    ((Button)v).setText("disable left drag");
                }
                ((DragView)view.findViewById(R.id.swipe_example_2_left_drag)).setDraggable(state);

            }
        });
        view.findViewById(R.id.btn_disable_right_drag).setOnClickListener(new View.OnClickListener() {
            private boolean state = true;
            public void onClick(View v) {
                if ( state ) {
                    state = false;
                    ((Button)v).setText("enable right drag");

                } else {
                    state = true;
                    ((Button)v).setText("disable right drag");
                }
                ((DragView)view.findViewById(R.id.swipe_example_2_right_drag)).setDraggable(state);

            }
        });
    }
}

File: app/src/main/java/kaufland/com/demo/example/EasyExample.kt
package kaufland.com.demo.example

import android.app.Fragment
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import android.widget.Toast

import kaufland.com.demo.R
import kaufland.com.swipelibrary.SwipeLayout
import kaufland.com.swipelibrary.SwipeState
import kotlinx.android.synthetic.main.easy_example_fragment.*
import kotlinx.android.synthetic.main.match_parent_example.*

/**
 * Created by sbra0902 on 30.03.17.
 */

class EasyExample : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle): View? {
        return inflater.inflate(R.layout.easy_example_fragment, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        setupExample2()

        btn_add.setOnClickListener { text_quantity.text = (Integer.valueOf(text_quantity.text.toString()) + 1).toString() }
        btn_minus.setOnClickListener { text_quantity.text = (Integer.valueOf(text_quantity.text.toString()) - 1).toString() }

    }

    private fun setupExample2() {
        swipe_example_2.setSwipeListener(object : SwipeLayout.SwipeListener {
            override fun onSwipeOpened(openedDragView: SwipeState.DragViewState, isFullSwipe: Boolean) {
                Toast.makeText(activity, "onSwipeOpened " + openedDragView.name + isFullSwipe.toString(), Toast.LENGTH_SHORT).show()
            }

            override fun onSwipeClosed(dragViewState: SwipeState.DragViewState) {
                Toast.makeText(activity, "onSwipeOpened " + dragViewState.name, Toast.LENGTH_SHORT).show()
            }

            override fun onBounce(dragViewState: SwipeState.DragViewState) {
                Toast.makeText(activity, "onBounce " + dragViewState.name, Toast.LENGTH_SHORT).show()
            }
        })
        btn_close.setOnClickListener { swipe_example_2.closeSwipe() }

        surface_view_example2.setOnClickListener { Toast.makeText(activity, "Clicked", Toast.LENGTH_SHORT).show() }
        btn_open_left.setOnClickListener { swipe_example_2.openSwipe(SwipeLayout.LEFT_DRAG_VIEW) }
        btn_open_right.setOnClickListener { swipe_example_2.openSwipe(SwipeLayout.RIGHT_DRAG_VIEW) }
        btn_disable_left_drag.setOnClickListener(object : View.OnClickListener {
            private var state = true
            override fun onClick(v: View) {
                if (state) {
                    state = false
                    (v as Button).text = "enable left drag"

                } else {
                    state = true
                    (v as Button).text = "disable left drag"
                }
                swipe_example_2_left_drag.isDraggable = state

            }
        })
        btn_disable_right_drag.setOnClickListener(object : View.OnClickListener {
            private var state = true
            override fun onClick(v: View) {
                if (state) {
                    state = false
                    (v as Button).text = "enable right drag"

                } else {
                    state = true
                    (v as Button).text = "disable right drag"
                }
                swipe_example_2_right_drag.isDraggable = state

            }
        })
    }
}
